GVFS/GVFS/CommandLine/UpgradeVerb.cs

```diff
@@ -73,14 +74,18 @@ private bool TryInitializeUpgrader()
 
                     this.tracer = jsonTracer;
                     this.prerunChecker = new InstallerPreRunChecker(this.tracer, this.Confirmed ? GVFSConstants.UpgradeVerbMessages.GVFSUpgradeConfirm : GVFSConstants.UpgradeVerbMessages.GVFSUpgrade);
-                    this.upgrader = new ProductUpgrader(ProcessHelper.GetCurrentProcessVersion(), this.tracer);
+                    this.upgrader = ProductUpgrader.CreateUpgrader(this.tracer, out error);
+                }
+                else
+                {
+                    error = null;

```

Comment:
Let's set this immediately on entering the method rather than having an else clause here.